### ISSUE TITLE
Inconsistent handling of commandline arguments

### ISSUE THREAD

**Issue by dokutan** [bakpakin/Fennel#428]
When running a Fennel script with commandline arguments, some arguments are treated as arguments to the interpreter.

## The problem
Take the the following script (arg.fnl) for example:
```
#!/usr/bin/env fennel
(while (. arg 1)
  (print (. arg 1))
  (table.remove arg 1))
```
When called as ``./arg.fnl -l`` or ``fennel arg.fnl -l``, the following error occurs:
```
runtime error: bad argument #1 to 'open' (string expected, got nil)
stack traceback:
  [C]: in function 'open'
  /usr/bin/fennel:5050: in function ?
  [C]: in function 'xpcall'
  /usr/bin/fennel:5709: in function 'dosafely'
  /usr/bin/fennel:5748: in function 'handle_load'
  /usr/bin/fennel:5801: in main chunk
  [C]: in ?
```
However ``./arg.fnl --version`` or ``fennel arg.fnl --version`` results in:
```
--version
```

## Comparison with Lua
Compare this behaviour to Lua, when arg.fnl is compiled to arg.lua:
```
while arg[1] do
  print(arg[1])
  table.remove(arg, 1)
end
return nil
```
``lua arg.lua -l`` prints ``-l``, ``lua -l arg.lua`` treats ``-l`` as an argument to Lua, resulting in an error.

## Intended behaviour
In my opionion Fennel should only accept arguments that appear before the first filename, all other arguments should be passed to the script. When calling a script directly as an executable no arguments should be handled by Fennel. This would make the behaviour of Fennel identical to Lua.

**Comment by technomancy** [comment]
Good catch; I agree.

Would you be up for submitting a fix?

**Comment by dokutan** [comment]
> Would you be up for submitting a fix?

Sure, i have opened #429.

**Comment by technomancy** [comment]
Fixed in e84c167
